IllegalStateException on Toolbar

```
Caused by: java.lang.IllegalStateException: 
  at androidx.appcompat.app.AppCompatDelegateImpl.setSupportActionBar (AppCompatDelegateImpl.java:345)
  at androidx.appcompat.app.AppCompatActivity.setSupportActionBar (AppCompatActivity.java:130)
  at com.owncloud.android.ui.activity.ToolbarActivity.setupToolbar (ToolbarActivity.java:69)
  at com.owncloud.android.ui.activity.ToolbarActivity.setupToolbar (ToolbarActivity.java:99)
  at com.owncloud.android.ui.activity.FileDisplayActivity.onCreate (FileDisplayActivity.java:244)
  at android.app.Activity.performCreate (Activity.java:7326)
  at android.app.Activity.performCreate (Activity.java:7317)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1271)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3066)
```

This is the IllegalStateException:
`"This Activity already has an action bar supplied by the window decor. Do not request Window.FEATURE_SUPPORT_ACTION_BAR and set windowActionBar to false in your theme to use a Toolbar instead."`

Seems to happen quite often, starting with 3.6.1 RC2.
Unfortunately I do not have any infos how to reproduce it.

--- a/src/androidTest/java/com/owncloud/android/ui/activity/FileDisplayActivityTest.java
+++ b/src/androidTest/java/com/owncloud/android/ui/activity/FileDisplayActivityTest.java
@@ -1,0 +1,23 @@
+package com.owncloud.android.ui.activity;
+
+import com.owncloud.android.AbstractIT;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.rule.GrantPermissionRule;
+
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+
+public class FileDisplayActivityTest extends AbstractIT {
+
+    @Rule public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(WRITE_EXTERNAL_STORAGE);
+
+    @Test
+    public void testSetupToolbar() {
+        try (ActivityScenario<FileDisplayActivity> scenario = ActivityScenario.launch(FileDisplayActivity.class)) {
+            scenario.recreate();
+        }
+    }
+}

--- a/src/main/java/com/owncloud/android/authentication/PassCodeManager.java
+++ b/src/main/java/com/owncloud/android/authentication/PassCodeManager.java
@@ -101,7 +101,7 @@ public final class PassCodeManager {
             i.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
             activity.startActivityForResult(i, PASSCODE_ACTIVITY);
         } else {
-            if (preferences.getLockTimestamp() != 0) {
+            if (!askedForPin && preferences.getLockTimestamp() != 0) {
                 preferences.setLockTimestamp(System.currentTimeMillis());
             }
         }

--- a/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java
+++ b/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.java
@@ -228,10 +228,11 @@ public class FileDisplayActivity extends FileActivity
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         Log_OC.v(TAG, "onCreate() start");
-        super.onCreate(savedInstanceState); // this calls onAccountChanged() when ownCloud Account is valid
 
         // Set the default theme to replace the launch screen theme.
         setTheme(R.style.Theme_ownCloud_Toolbar_Drawer);
+
+        super.onCreate(savedInstanceState); // this calls onAccountChanged() when ownCloud Account is valid
 
         /// Load of saved instance state
         if (savedInstanceState != null) {

